src/main/scala/com/atamaniv/tailrec/RetryImpl.scala

```diff
@@ -0,0 +1,21 @@
+package com.atamaniv.tailrec
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+object RetryImpl {
+
+  //Added Option here, for me it is more logical, if condition is not reached then None is returned.
+  final def retry[A](block: () => Future[Option[A]], acceptResult: A => Boolean, retries: List[FiniteDuration]): Future[Option[A]] = {
```

Comment:
Again, the task **required** to return the last result in case all retries are exceeded, your version looses last result.
Think of retrying http calls (I've explained that a lot of times), you will likely want to get last failed result, parse it and log it somewhere, maybe send metrics or do possible other action depending on the last status code. You will also want to keep retry function simple and not pollute it with things that it shouldn't do.